src/scanpy/preprocessing/_simple.py

```diff
@@ -678,7 +707,6 @@ def regress_out(
 
     if issparse(X):
         logg.info("    sparse input is densified and may lead to high memory use")
-        X = X.toarray()
```

Comment:
As a follow up probably worth replacing any instance of this in scanpy with the approporiate numba'ed version

Reply:
Yes we will do that. However that will be a separate PR.